busted/outputHandlers/junit.lua

```diff
@@ -101,10 +127,37 @@ return function(options)
     end
   end
 
+  local function get_junit_info(path)
+    local junit_report_package_name, test_file_name
+
+    if string.match(path, "/") or string.match(path, "\\") then
+      -- Compatible with Windows platform
+      junit_report_package_name, test_file_name = path:match("(.-)[\\/]+([^\\/]+)$")
+    else
+      test_file_name = path
+      junit_report_package_name = ""
+    end
+
+    return junit_report_package_name, test_file_name
+  end
+
   handler.testStart = function(element, parent)
+    local junit_classname
+    local test_case_full_name = handler.getFullName(element)
+    local junit_report_package_name, test_file_name = get_junit_info(element.trace.short_src)
+    -- Jenkins CI Junit Plugin use the last one . to distinguish between package name and class name.
+    local junit_class_name = string.gsub(test_file_name, "%.", "_")
+
+    if junit_report_package_name ~= "" then
+      junit_classname = junit_report_package_name .. "." .. junit_class_name ..":" .. element.trace.currentline
+    else
+      junit_classname = junit_class_name ..":" .. element.trace.currentline
+    end
+
     testcase_node = xml.new('testcase', {
-      classname = element.trace.short_src .. ':' .. element.trace.currentline,
-      name = handler.getFullName(element),
+      -- junit report uses package names and class names to structurally display result.
+      classname = junit_classname,
```

Comment:
Reason for modification reference:

https://github.com/jenkinsci/junit-plugin/blob/67297b3bcae8ec8422b1b9a965595f7187fd9318/src/main/java/hudson/tasks/junit/CaseResult.java#L182